jobs/mysql_backup_job.go

```diff
@@ -154,25 +163,14 @@ func (j *MySQLBackupJob) Run(ctx context.Context, send Send) (rerr error) {
 			return wrapError(err)
 		}
 
-		defer func() {
-			err := xbcloudCmd.Wait()
-			if err == nil {
-				return
-			}
-
-			if rerr != nil {
-				rerr = errors.Wrapf(rerr, "xbcloud wait error: %s", err)
-			} else {
-				rerr = wrapError(err)
-			}
-		}()
-	}
-
-	if err := xtrabackupCmd.Start(); err != nil {
-		return wrapError(err)
+		if err := xbcloudCmd.Wait(); err != nil {
+			_ = xtrabackupStdout.Close()
+			return wrapError(err)
+		}
 	}
 
-	if err := xtrabackupCmd.Wait(); err != nil {
+	err = <-xtrabackupErr
```

Comment:
it won't come to this line if xbcloudcmd will fail, so we will have one goroutine which will hang.

Reply:
I tried it again with `io.Pipe` and goroutine exits properly.